packages/eslint-plugin/tests/rules/explicit-function-return-type.test.ts

```diff
@@ -468,6 +468,27 @@ const func = (value: number) => x as const satisfies number;
     },
     {
       code: `
+const func = (value: number) => x as const satisfies 10 satisfies number;
```

Comment:
fun fact, this code is illegal TS, since `as const` cannot be used on identifiers. Sometimes we intentionally write test cases with illegal TS, but lately we've generally been trying to avoid doing so (or we try to provide a `@ts-expect-error` comment to indicate we're intentionally testing illegal TS). Let's use an alternative here (and in the explicit-module-boundary test case too) that isn't an error since we can easily swap it out with one? See https://github.com/typescript-eslint/typescript-eslint/pull/10315#discussion_r1836146131 for a (surprisingly tedious to craft) example of a legal nested-`satisfies` `as const` pattern.

Reply:
Sure, I've updated the tests I added to be valid TypeScript 👍